test/e2e/module-layer/index.test.ts

```diff
@@ -60,45 +61,60 @@ createNextDescribe(
       }
     }
 
-    describe('no server-only in server targets', () => {
-      const middlewareFile = 'middleware.js'
-      // const pagesApiFile = 'pages/api/hello.js'
-      let middlewareContent = ''
-      // let pagesApiContent = ''
+    describe('with server-only in server targets', () => {
+      runTests()
+    })
 
-      beforeAll(async () => {
-        await next.stop()
+    // Should error for using mixed (with client-only) in server targets
+    if (isNextDev) {
+      describe('no server-only in server targets', () => {
+        const middlewareFile = 'middleware.js'
+        // const pagesApiFile = 'pages/api/hello.js'
+        let middlewareContent = ''
+        // let pagesApiContent = ''
 
-        middlewareContent = await next.readFile(middlewareFile)
-        // pagesApiContent = await next.readFile(pagesApiFile)
+        beforeAll(async () => {
+          await next.stop()
 
-        await next.patchFile(
-          middlewareFile,
-          middlewareContent
-            .replace("import 'server-only'", "// import 'server-only'")
-            .replace("// import './lib/mixed-lib'", "import './lib/mixed-lib'")
-        )
+          middlewareContent = await next.readFile(middlewareFile)
+          // pagesApiContent = await next.readFile(pagesApiFile)
 
-        // await next.patchFile(
-        //   pagesApiFile,
-        //   pagesApiContent
-        //     .replace("import 'server-only'", "// import 'server-only'")
-        //     .replace(
-        //       "// import '../../lib/mixed-lib'",
-        //       "import '../../lib/mixed-lib'"
-        //     )
-        // )
+          await next.patchFile(
+            middlewareFile,
+            middlewareContent
+              // .replace("import 'server-only'", "// import 'server-only'")
+              .replace(
+                "// import './lib/mixed-lib'",
+                "import './lib/mixed-lib'"
+              )
+          )
 
-        await next.start()
-      })
-      afterAll(async () => {
-        await next.patchFile(middlewareFile, middlewareContent)
-        // await next.patchFile(pagesApiFile, pagesApiContent)
+          // await next.patchFile(
```

Comment:
leave these comments here, which could be enabled later once we apply the same rule to pages/api